detox/src/devices/DeviceRegistry.js

```diff
@@ -0,0 +1,75 @@
+const fs = require('fs');
+const plockfile = require('proper-lockfile');
+const _ = require('lodash');
+const retry = require('../utils/retry');
+const LOCK_FILE = './device.registry.state.lock';
```

Comment:
To support multiple Detox runs (e.g. in a non virtualized CI machine) LOCK_FILE should be added globally. We already create a `~/Library/Detox/` when compiling the framework, I think LOCK_FILE should be there.